src/sage/graphs/graph.py

```diff
@@ -10228,6 +9612,13 @@ def bipartite_double(self, extended=False):
     from sage.graphs.graph_coloring import fractional_chromatic_number
     from sage.graphs.graph_coloring import fractional_chromatic_index
     from sage.graphs.hyperbolicity import hyperbolicity
+    has_perfect_matching = LazyImport('sage.graphs.matching', 'has_perfect_matching', at_startup=True)
+    is_bicritical = LazyImport('sage.graphs.matching', 'is_bicritical', at_startup=True)
+    is_factor_critical = LazyImport('sage.graphs.matching', 'is_factor_critical', at_startup=True)
+    is_matching_covered = LazyImport('sage.graphs.matching', 'is_matching_covered', at_startup=True)
+    matching = LazyImport('sage.graphs.matching', 'matching', at_startup=True)
+    perfect_matchings = LazyImport('sage.graphs.matching', 'perfect_matchings', at_startup=True)
+    M_alternating_even_mark = LazyImport('sage.graphs.matching', 'M_alternating_even_mark', at_startup=True)
```

Comment:
why should this method be exposed in Graph ? is it needed for other purposes than where it is already used ?

Reply:
some methods could be moved, but in a future PR.